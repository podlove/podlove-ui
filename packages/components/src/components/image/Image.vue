<script lang="ts" setup>
import { ref } from 'vue';

defineProps<{ url: string; alt?: string }>();

const emit = defineEmits(['load', 'error']);

const loaded = ref(false);
const error = ref(false);

const loadHandler = (event: Event) => {
  loaded.value = true;
  emit('load', event);
};

const errorHandler = (event: Event) => {
  error.value = true;
  emit('error', event);
};
</script>

<template>
  <div class="relative">
    <transition name="fade">
      <div
        v-if="!loaded"
        class="podlove-component-image-cover absolute top-0 left-0 w-full h-full"
      ></div>
    </transition>
    <img
      v-if="url && !error"
      class="max-h-full w-auto"
      :class="{ 'h-0': !loaded }"
      :src="url"
      :alt="alt"
      @error="errorHandler"
      @load="loadHandler"
    />
  </div>
</template>

<style lang="postcss" scoped>
.fade-enter-active,
.fade-leave-active {
  transition: opacity 600ms;
  opacity: 1;
}
.fade-enter,
.fade-leave-to {
  opacity: 0;
}

.podlove-component-image-cover {
  background-color: var(--podlove-component-image-background, var(--podlove-components-background));
}
</style>
