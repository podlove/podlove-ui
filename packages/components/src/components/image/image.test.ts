import { describe, test, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import Image from './Image.vue';

const props = {
  url: 'http://foo.bar',
  alt: 'alternative title'
}

describe('Image', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(Image, { props });
      expect(wrapper.vm).toBeTruthy();
    });
  });

  describe('props', () => {
    test('should render', async () => {
      const wrapper = await mount(Image, { props });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
