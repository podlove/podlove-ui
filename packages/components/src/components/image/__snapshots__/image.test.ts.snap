// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image > props > should render 1`] = `
<div
  class="relative"
  data-v-2f3a6608=""
>
  <transition-stub
    appear="false"
    css="true"
    data-v-2f3a6608=""
    name="fade"
    persisted="false"
  >
    <div
      class="podlove-component-image-cover absolute top-0 left-0 w-full h-full"
      data-v-2f3a6608=""
    />
  </transition-stub>
  <img
    alt="alternative title"
    class="max-h-full w-auto h-0"
    data-v-2f3a6608=""
    src="http://foo.bar"
  />
</div>
`;

exports[`Image > props > should render with prop alt 1`] = `
<div
  class="relative"
  data-v-2f3a6608=""
>
  <transition-stub
    appear="false"
    css="true"
    data-v-2f3a6608=""
    name="fade"
    persisted="false"
  >
    <div
      class="podlove-component-image-cover absolute top-0 left-0 w-full h-full"
      data-v-2f3a6608=""
    />
  </transition-stub>
  <!--v-if-->
</div>
`;

exports[`Image > props > should render with prop color 1`] = `
<div
  class="relative"
  color="#fff"
  data-v-2f3a6608=""
>
  <transition-stub
    appear="false"
    css="true"
    data-v-2f3a6608=""
    name="fade"
    persisted="false"
  >
    <div
      class="podlove-component-image-cover absolute top-0 left-0 w-full h-full"
      data-v-2f3a6608=""
    />
  </transition-stub>
  <!--v-if-->
</div>
`;

exports[`Image > props > should render with prop url 1`] = `
<div
  class="relative"
  data-v-2f3a6608=""
>
  <transition-stub
    appear="false"
    css="true"
    data-v-2f3a6608=""
    name="fade"
    persisted="false"
  >
    <div
      class="podlove-component-image-cover absolute top-0 left-0 w-full h-full"
      data-v-2f3a6608=""
    />
  </transition-stub>
  <img
    class="max-h-full w-auto h-0"
    data-v-2f3a6608=""
    src="http://foo.bar"
  />
</div>
`;

exports[`Image > props > should render without props 1`] = `
<div
  class="relative"
  data-v-2f3a6608=""
>
  <transition-stub
    appear="false"
    css="true"
    data-v-2f3a6608=""
    name="fade"
    persisted="false"
  >
    <div
      class="podlove-component-image-cover absolute top-0 left-0 w-full h-full"
      data-v-2f3a6608=""
    />
  </transition-stub>
  <!--v-if-->
</div>
`;
